python/tank/template.py

```diff
@@ -439,12 +439,71 @@ def parent(self):
+                         default value of None, paths will be created to match the 
+                         current operating system. If you pass in a platform constant
+                         (e.g. Template.PLATFORM_XXX), paths will be generating to 
+                         target that specific platform.
+
+        :returns: Full path, matching the template with the given fields inserted.
+        """        
+        relative_path = super(TemplatePath, self)._apply_fields(fields, ignore_types, platform)
+        
+        if platform is None:
+            # return the current OS platform's path
+            return os.path.join(self.root_path, relative_path) if relative_path else self.root_path
+    
+        else:
+            # caller has requested a path for another OS
+            
+            if self._root_paths_all_os is None:
+                # it's possible that the additional os paths are not set for a template
+                # object (mainly because of backwards compatibility reasons) and in this case
+                # we cannot compute the path.
+                raise TankError("Template %s cannot resolve path for operating system '%s' - "
+                                "it was instantiated in a mode which only supports the resolving "
+                                "of current operating system paths." % (self, platform))
+            
+            PATH_LOOKUP = {self.PLATFORM_LINUX: "linux2", 
+                           self.PLATFORM_MAC: "darwin", 
+                           self.PLATFORM_WINDOWS: "win32"}
+            
+            platform_root_path = self._root_paths_all_os.get(PATH_LOOKUP[platform])
+            
+            if platform_root_path is None:
+                # either the platform is undefined or unknown
+                raise TankError("Cannot resolve path for unknown operating system '%s'! Please ensure "
+                                "that you have a valid storage set up for this platform." % platform)
+            
+            elif platform == Template.PLATFORM_WINDOWS:
+                # use backslashes for windows
+                if relative_path:
+                    return "%s\\%s" % (platform_root_path, relative_path.replace(os.sep, "\\"))
+                else:
+                    # not path generated - just return the root path
+                    return platform_root_path
+            
+            elif platform in [Template.PLATFORM_LINUX, Template.PLATFORM_MAC]:
+                # unix-like plaforms - use slashes
+                if relative_path:
+                    return "%s/%s" % (platform_root_path, relative_path.replace(os.sep, "/"))
+                else:
+                    # not path generated - just return the root path 
+                    return platform_root_path
+            
+            else:
+                raise TankError("Cannot evaluate path. Unsupported platform '%s'." % platform)
```

Comment:
no lookup would be available here but if platform were a string instead of an integer then this message would be more meaningful.